File: Chain/python/Request.py
from enum import Enum

class Priority(Enum):
    BASIC = 0
    INTERMEDIATE = 1
    CRITICAL = 2
    FATAL = 3
    

class Request:
    priority: Priority
    
    def __init__(self, priority: Priority):
        self.priority = priority
        
    
    def get_priority(self) -> Priority:
        return self.priority

File: Chain/ts/Request.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Request = exports.Priority = void 0;
var Priority;
(function (Priority) {
    Priority[Priority["BASIC"] = 0] = "BASIC";
    Priority[Priority["INTERMEDIATE"] = 1] = "INTERMEDIATE";
    Priority[Priority["CRITICAL"] = 2] = "CRITICAL";
})(Priority || (exports.Priority = Priority = {}));
class Request {
    constructor(priority) {
        this.priority = priority;
    }
    getPriority() {
        return this.priority;
    }
}
exports.Request = Request;
